endpoint.go

```diff
@@ -289,10 +290,29 @@ func (ep *endpoint) Join(sbox Sandbox, options ...EndpointOption) error {
 		return err
 	}
 
+	sb.Lock()
+	heap.Push(&sb.endpoints, ep)
+	sb.Unlock()
+	defer func() {
+		if err != nil {
+			var (
```

Comment:
Yes. I decided to take this approach to be consistent with the heap Remove code elsewhere. But I can certainly make it simple.